LifeCycleAssessment_Engine/Compute/EvaluateEnvironmentalProductDeclaration.cs

```diff
@@ -49,32 +42,55 @@ public static partial class Compute
         [Output("result", "A LifeCycleElementResult that contains the LifeCycleAssessment data for the input object.")]
         public static LifeCycleAssessmentElementResult EvaluateEnvironmentalProductDeclarationPerObject(IElementM elementM, EnvironmentalProductDeclarationField field = EnvironmentalProductDeclarationField.GlobalWarmingPotential)
         {
-            QuantityType qt = elementM.GetFragmentQuantityType();
+            double value = 0;
+            GlobalWarmingPotentialResult gwpr = null;
 
-            switch (qt)
+            List<QuantityType> qts = elementM.GetQuantityType();
+
+            qts = qts.Distinct().ToList();
+
+            foreach (QuantityType qt in qts)
             {
-                case QuantityType.Undefined:
-                    BH.Engine.Reflection.Compute.RecordError("The object's EPD QuantityType is Undefined and cannot be evaluated.");
-                    return null;
-                case QuantityType.Item:
-                    BH.Engine.Reflection.Compute.RecordError("Length QuantityType is currently not supported. Try a different EPD with QuantityType values of either Area, Volume, or Mass.");
-                    return null;
-                case QuantityType.Length:
-                    BH.Engine.Reflection.Compute.RecordError("Length QuantityType is currently not supported. Try a different EPD with QuantityType values of either Area, Volume, or Mass.");
-                    return null;
-                case QuantityType.Area:
-                    BH.Engine.Reflection.Compute.RecordNote("Evaluating object type: " + elementM.GetType() + " based on EPD Area QuantityType.");
-                    return EvaluateEnvironmentalProductDeclarationByArea(elementM, field);
-                case QuantityType.Volume:
-                    BH.Engine.Reflection.Compute.RecordNote("Evaluating object type: " + elementM.GetType() + " based on EPD Volume QuantityType.");
-                    return EvaluateEnvironmentalProductDeclarationByVolume(elementM, field);
-                case QuantityType.Mass:
-                    BH.Engine.Reflection.Compute.RecordNote("Evaluating object type: " + elementM.GetType() + " based on EPD Mass QuantityType.");
-                    return EvaluateEnvironmentalProductDeclarationByMass(elementM, field);
-                default:
-                    BH.Engine.Reflection.Compute.RecordWarning("The object you have provided does not contain an EPD Material Fragment.");
-                    return null;
+                switch (qt)
```

Comment:
Switch case handles making sure each element's QT is calling the appropriate method... before it gets to the method itself.